### PR TITLE
chore/update nx and angular 13

### PR DESCRIPTION
**Pull request by the-ult**
- Upgrade the project to NX 13.2 with Angular 13
- Setup [NX per project configuration](https://blog.nrwl.io/per-project-configuration-storybook-support-for-angular-12-auto-refresh-for-dependency-graph-and-bcd2d1e06658)
- Update other dependencies
- improve build targets and project tsconfig build settings

## TODO (_might need some help with these_)
- [x] Properly setup build targets with `@nrwl/angular:package` and use `tsconfig.lib.prod.json` configuration for production
  ```json
    "configurations": {
        "production": {
          "tsConfig": "projects/testing-library/tsconfig.lib.prod.json"
        },
        "development": {
          "tsConfig": "projects/testing-library/tsconfig.lib.json"
        }
      },
  ```
- [ ] Breaking changes to README/Changelog?
- [ ] Update Node version to LTS (16?) @timdeschryver it seems to be at 12 atm.

fixes: #258

### CODE REVIEW

File: apps/example-app-karma/src/test.ts [hunk 1]
```diff
@@ -11,7 +11,9 @@ beforeAll(() => {
 declare const require: any;
 
 // First, initialize the Angular testing environment.
-getTestBed().initTestEnvironment(BrowserDynamicTestingModule, platformBrowserDynamicTesting());
+getTestBed().initTestEnvironment(BrowserDynamicTestingModule, platformBrowserDynamicTesting(), {
+  teardown: { destroyAfterEach: false },
```
Comment (the-ult) [inline apps/example-app-karma/src/test.ts:15]: automatically set to false. We might want to set it to true?

File: projects/jest-utils/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{
```
Comment (the-ult) [inline projects/jest-utils/tsconfig.json:1]: I changed the way the project `tsconfig.*.json` are setup. So they are more in line in what a new NX project would generate a lib.

File: apps/example-app-karma/src/test.ts (hunk 1 above)
Comment (timdeschryver) [reply apps/example-app-karma/src/test.ts:15]: Yes, you can remove this config - I would like to use the default settings.

**Review by timdeschryver — COMMENTED** [review]
Thanks @the-ult, I will create a `beta` branch for this and  change the target branch later this week. This way we can test it out, and add some more breaking changes before we release the new version.

- `  "executor": "@nrwl/angular:package",`: yes, you can update it
- `Should we update the peerDependencies to >= 13.0.0`: yes
- ` config tot Render for teardown`: let's do this in another PR after this is merged
- `Breaking changes to README/Changelog?`: these will be automatic 🙂

File: package.json [hunk 3]
```diff
@@ -2,16 +2,16 @@
   "name": "@testing-library/angular-app",
   "version": "0.0.0-semantically-released",
   "scripts": {
-    "ng": "ng",
+    "ng": "nx",
     "nx": "nx",
-    "postinstall": "ngcc",
-    "start": "ng serve",
+    "postinstall": "node ./decorate-angular-cli.js && ngcc --properties es2015 browser module main",
```
Comment (timdeschryver) [inline package.json:7]: Do we still need this?

File: projects/testing-library/project.json [hunk 4]
```diff
@@ -0,0 +1,60 @@
+{
+  "projectType": "library",
+  "root": "projects/testing-library",
+  "sourceRoot": "projects/testing-library/src",
+  "prefix": "lib",
+  "targets": {
+    "build-package": {
+      "executor": "@nrwl/angular:package",
+      "outputs": ["dist/@testing-library/angular"],
```
[the remark this answers is not in the document]
Comment (timdeschryver) [reply projects/testing-library/project.json:9]: Yes, I believe both output paths are correct.

File: package.json (hunk 3 above)
Comment (the-ult) [reply package.json:7]: Yes, in a freshly create NX workspace, this is the default.

**Review by timdeschryver — APPROVED** [review]

**Comment by timdeschryver** [comment]
Let's get this rolling! 🥳

**Comment by timdeschryver** [comment]
@all-contributors please add @the-ult for maintenance

**Pull request merged**